index.js

```diff
@@ -1,18 +1,27 @@
 const { codeFrameColumns } = require("@babel/code-frame");
 const Worker = require("jest-worker").default;
-const serialize = require("serialize-javascript");
+const { generate } = require("escodegen");
+const lave = require("lave");
 
 function terser(userOptions = {}) {
   if (userOptions.sourceMap != null) {
     throw Error("sourceMap option is removed, use sourcemap instead");
   }
 
-  const minifierOptions = serialize(
-    Object.assign({}, userOptions, {
-      sourceMap: userOptions.sourcemap !== false,
-      sourcemap: undefined,
-      numWorkers: undefined
-    })
+  const normalizedOptions = {
+    ...userOptions,
+    ...{ sourceMap: userOptions.sourcemap !== false }
+  };
+
+  for (let key of ["sourcemap", "numWorkers"]) {
+    if (normalizedOptions.hasOwnProperty(key)) {
```

Comment:
It seems like hasOwnProperty is not required with for of loop.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#Difference_between_for...of_and_for...in

Reply:
Could you be more specific please?

I'm not looping over normalized options. I've made the loop over `sourcemap` and `numWorkers` because I cannot set them to undefined in the merge above. As said earlier any undefined props will be serialized.

Of course we could unroll this loop into two seperate deletes checking with two `hasOwnProps` but that would be code duplication and this way is more generic if there will be more properties we'd have to delete.

Reply:
Ah, sorry. You are right.